### PR TITLE
06307 - Events Support Noncontiguous Node

### LINKED ISSUE #6307: Noncontiguous NodeId Batch - Events and Transaction Handling
Use NodeId API from AddressBook

For each batch, these are the changes that need to be made. 
- NodeId - All methods returning a long node id now return NodeId
- NodeId - All methods consuming a long node id now consume NodeId
- NodeId - Change usage of int node ids as index of arrays/lists to  Map<NodeId, ?>
- NodeId - Complete translation of all other usage of int/long node ids to NodeId
- Ensure unit tests work with noncontiguous NodeId

Files:
- ~~ConsensusImpl~~
- ~~ConsensusHashManager~~
- CriticalQuorumImpl
- EventCreatorThread
- EventCreator
- EventTaskCreator
- EventImpl - minus Consensus Related Functions
- TransactionPool
- Unit Tests (Related)

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:
Apologies to the reviewers.  When you change a core API, the cascade of files that need to be updated is unfortunate. 

Fixes #6307

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/CriticalQuorumTest.java [hunk 1]
```diff
@@ -88,10 +91,10 @@ private static EventImpl buildSimpleEvent(final long creatorId, final long round
     /**
      * Build and initialize a map for tracking event counts.
      */
-    private static Map<Integer, Integer> buildEventCountMap(final AddressBook addressBook) {
-        final Map<Integer, Integer> eventCountMap = new HashMap<>();
-        for (int nodeId = 0; nodeId < addressBook.getSize(); nodeId++) {
-            eventCountMap.put(nodeId, 0);
+    private static Map<NodeId, Integer> buildEventCountMap(final AddressBook addressBook) {
+        final Map<NodeId, Integer> eventCountMap = new HashMap<>();
+        for (Address address : addressBook) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/CriticalQuorumTest.java:96]:
```suggestion
        for (final Address address : addressBook) {
```

**Review by cody-littley — APPROVED** [review]
Please seek Lazar's feedback on this before merging, since this will likely generate merge conflicts for him. He may do a full review, or he may glance at it and say 👍, it's up to him.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/chatter/protocol/messages/ChatterEventDescriptor.java [hunk 2]
```diff
@@ -74,7 +76,9 @@ public long getClassId() {
     @Override
     public void serialize(final SerializableDataOutputStream out) throws IOException {
         out.writeSerializable(hash, false);
-        out.writeLong(creator);
+        // FUTURE WORK: The creator should be a selfSerializable NodeId at some point.
+        // Changing the event format may require a HIP.  The old format is preserved for now.
+        out.writeLong(creator.id());
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/chatter/protocol/messages/ChatterEventDescriptor.java:81]: changed implementation to update class version and use SelfSerializable NodeId.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/chatter/protocol/messages/ChatterEventDescriptor.java [hunk 3]
```diff
@@ -84,7 +88,14 @@ public void serialize(final SerializableDataOutputStream out) throws IOException
     @Override
     public void deserialize(final SerializableDataInputStream in, final int version) throws IOException {
         hash = in.readSerializable(false, Hash::new);
-        creator = in.readLong();
+        // FUTURE WORK: The creator should be a selfSerializable NodeId at some point.
+        // Changing the event format may require a HIP.  The old format is preserved for now.
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/chatter/protocol/messages/ChatterEventDescriptor.java:92]: changed implementation to update class version and use SelfSerializable NodeId.  removed comments.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java [hunk 4]
```diff
@@ -222,7 +226,14 @@ public void deserialize(
         } else {
             softwareVersion = SoftwareVersion.NO_VERSION;
         }
-        creatorId = in.readLong();
+        // FUTURE WORK: The creatorId should be a selfSerializable NodeId at some point.
+        // Changing the event format may require a HIP.  The old format is preserved for now.
+        final long serializedCreatorId = in.readLong();
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java:231]: Maybe add a static deserialize method to `NodeId` with this logic

**Review by lpetrovic05 — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java (hunk 4 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java]: Added

**Review by cody-littley — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

**Pull request merged**